File: shopsphere-frontend/src/services/reviewService.js
import axiosInstance from "../api/axios";

export const getReviewsByProduct = (productId) => {
  return axiosInstance.get(
    `/reviews/product/${productId}`
  );
};

export const addReview = (reviewData) => {
  return axiosInstance.post(
    "/reviews",
    reviewData
  );
};

export const updateReview = (
  reviewId,
  reviewData
) => {
  return axiosInstance.put(
    `/reviews/${reviewId}`,
    reviewData
  );
};

export const deleteReview = (reviewId) => {
  return axiosInstance.delete(
    `/reviews/${reviewId}`
  );
};

File: Backend/src/main/java/com/shopsphere/backend/service/ReviewService.java
package com.shopsphere.backend.service;

import com.shopsphere.backend.dto.review.ReviewRequest;
import com.shopsphere.backend.dto.review.ReviewResponse;

import java.util.List;

public interface ReviewService {

    ReviewResponse addReview(ReviewRequest request);

    List<ReviewResponse> getReviewsByProduct(Long productId);

    ReviewResponse updateReview(Long reviewId, ReviewRequest request);

    void deleteReview(Long reviewId);
}